src/Manager.js

```diff
@@ -1,19 +1,87 @@
-import Guest from "../src/Guest";
-
-class Manager extends Guest {
-  constructor(id, name, hotelInfo) {
-    super(id, name);
+class Manager {
+  constructor(id, name) {
     this.id = id;
     this.name = name;
-    this.hotelInfo = hotelInfo;
+    this.username = "manager";
+    this.password = "overlook2020";
+  }
+
+  calculateTodaysRevenue(bookingData, roomData, todaysDate) {
+    let todaysBookings = bookingData.filter(
+      booking => booking.date === todaysDate
+    );
+    let revenue = todaysBookings.reduce((revenue, bookedRoom) => {
+      roomData.forEach(room => {
+        if (room.number === bookedRoom.roomNumber) {
+          revenue += room.costPerNight;
+        }
+      });
+      return revenue;
+    }, 0);
+    return Number(revenue.toFixed(2));
+  }
+
+  searchGuest(guestName, guestData, bookingData, roomData) {
+    let guest = guestData.find(guest => guestName == guest.name);
+    let guestBookings = bookingData.filter(
+      booking => booking.userID === guest.id
+    );
+    let amountSpent = guestBookings.reduce((total, booking) => {
+      roomData.forEach(room => {
+        if (booking.roomNumber == room.number) {
+          total += room.costPerNight;
+        }
+      });
+      return total;
+    }, 0);
+    let guestInfo = {
+      bookings: guestBookings,
+      spending: amountSpent
+    };
+    return guestInfo;
+  }
+
+  calculatePercentOccupiedToday(bookingData, roomData, todaysDate) {
+    let numberOfBookingsToday = bookingData.filter(
+      booking => booking.date === todaysDate
+    ).length;
+    let percentageOfRoomsOccupiedToday =
+      (numberOfBookingsToday / roomData.length) * 100;
+    return percentageOfRoomsOccupiedToday;
   }
 
-  viewBooking(user) {
-    // search a user by name to view bookings and total amount spent
+  // TBT
+  bookRoom(userID, date, roomNumber) {
+    let bookingObject = {
+      userID: Number(userID),
+      date: date,
+      roomNumber: Number(roomNumber)
+    };
+    let url =
+      "https://fe-apps.herokuapp.com/api/v1/overlook/1904/bookings/bookings";
+    return fetch(url, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json"
+      },
+      body: JSON.stringify(bookingObject)
+    })
+      .then(response => console.log(response.json()))
+      .catch(err => console.log(err.message));
   }
 
-  deleteBooking() {
-    // Delete any upcoming room bookings for a user
+  // TBT
```

Comment:
Have you gotten this to work?  I haven't gotten to this part yet so I'm curious!